src/monitoring/blockListener.ts

```
@@ -0,0 +1,269 @@
+import WebSocket from 'ws';
+import { PublicKey } from '@solana/web3.js';
+import { BaseListener, EventCallback } from './baseListener'; // Assuming EventCallback is exported
+import { PumpEventProcessor, TokenInfo } from './blockEventProcessor'; // Assuming TokenInfo is exported
+import { SolanaClient } from '../core/client';
+import { Wallet } from '../core/wallet';
+import { pumpFunProgramId } from '../core/pubkeys'; // Default program ID
+
+// Placeholder for Trader
+interface Trader {
+  // Define methods and properties of Trader if known
+}
```

Biome (1.9.4)

[error] An empty interface is equivalent to {}.

Safe fix: Use a type alias instead.


(lint/suspicious/noEmptyInterface)